### PR TITLE
[hwasan] Add `__hwasan_get_tag_from_pointer`

### PR DESCRIPTION
**Pull request by vitalybuka**
This simplifies handling tags by user code. Now code does not need
to know bit size of tag and its position.

### CODE REVIEW

File: compiler-rt/test/sanitizer_common/sanitizer_specific.h
Comment (fmayer) [inline compiler-rt/test/sanitizer_common/sanitizer_specific.h:1]: Confused by this change. Why is this needed in this CL? If this is one of the purposes of the change, mention in the commit message.

File: compiler-rt/test/hwasan/TestCases/tag-ptr.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  std::set<void *> ptrs;
+  for (unsigned i = 0;; ++i) {
+    void *ptr = __hwasan_tag_pointer(p.get(), i);
+    if (!ptrs.insert(ptr).second)
+      break;
+    fprintf(stderr, "%p, %u, %u\n", ptr, i, __hwasan_get_tag_from_pointer(ptr));
+    assert(__hwasan_get_tag_from_pointer(ptr) == i);
+  }
+#ifdef __x86_64__
+  assert(ptrs.size() == 8);
```
Comment (thurstond) [inline compiler-rt/test/hwasan/TestCases/tag-ptr.cpp:20]: Is this assuming aliasing mode with 3-bit tags? If so, can the assumption be added to a `// REQUIRES` or similar?

**Review by thurstond — APPROVED** [review]

File: compiler-rt/test/hwasan/TestCases/tag-ptr.cpp (hunk 1 above)
Comment (vitalybuka) [reply compiler-rt/test/hwasan/TestCases/tag-ptr.cpp:20]: REQUIRES will disable the test completely. 
I guess we will update ifdef when we (or someone else) will have LAM
